RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h

```diff
@@ -26,14 +26,28 @@ namespace ticl {
         : algo_verbosity_(conf.getParameter<int>("algo_verbosity")) {}
     virtual ~PatternRecognitionAlgoBase(){};
 
-    virtual void makeTracksters(const edm::Event& ev,
-                                const edm::EventSetup& es,
-                                const std::vector<reco::CaloCluster>& layerClusters,
-                                const std::vector<float>& mask,
-                                const edm::ValueMap<float>& layerClustersTime,
-                                const TICLLayerTiles& tiles,
-                                const std::vector<TICLSeedingRegion>& regions,
-                                std::vector<Trackster>& result) = 0;
+    struct Inputs {
+      const edm::Event& ev;
+      const edm::EventSetup& es;
+      const std::vector<reco::CaloCluster>& layerClusters;
+      const std::vector<float>& mask;
+      const edm::ValueMap<float>& layerClustersTime;
+      const TICLLayerTiles& tiles;
+      const std::vector<TICLSeedingRegion>& regions;
+      std::vector<Trackster>& result;
```

Comment:
this is an output.

My request to aggregate was trying to address the fluid/growing part of the interface which is also apparently necessary only for a fraction of the derived algorithms.
"results" are somewhat distinct and non-const and should be used by every implementation. Better keep them separate.

BTW, is the ```result``` supposed to be empty on entry into ```makeTracksters(``` , if so, why not return by value?